packages/geo-diary-v2/src/components/ui/appointment-list/appointment-list.tsx

```diff
@@ -0,0 +1,88 @@
+import React from 'react'
+import { AppointmentModel, ListItemModel } from '@reapit/foundations-ts-definitions'
+import { Tile, Button, IconList, getTime } from '@reapit/elements'
+import { FaClock, FaStreetView, FaAddressCard } from 'react-icons/fa'
+import { ExtendedAppointmentModel } from '@/types/global'
+
+export const renderFooterItems = () => [
+  <>
+    <Button type="submit" variant="primary" onClick={() => null} disabled={false} loading={false} fullWidth={false}>
+      Details
+    </Button>
+    <Button type="submit" variant="primary" onClick={() => null} disabled={false} loading={false} fullWidth={false}>
+      Directions
+    </Button>
+  </>,
+]
+
+export type RenderIconItemsParams = {
+  appointment: ExtendedAppointmentModel
+  appointmentTypes: ListItemModel[]
+}
+
+export const renderIconItems = ({ appointment, appointmentTypes }: RenderIconItemsParams) => {
+  const line2 = appointment?.property?.address?.line2 ?? ''
+  const line3 = appointment?.property?.address?.line3 ?? ''
+  const line4 = appointment?.property?.address?.line4 ?? ''
+  const typeId = appointment?.typeId ?? ''
+  const postcode = appointment?.property?.address?.postcode ?? ''
+  const address = `${line2} ${line3} ${line4} ${postcode}`
+  const start = getTime(appointment?.start || '')
+  const end = getTime(appointment?.end || '')
+  const type = appointmentTypes.find(appointmentType => appointmentType.id === typeId)
+
+  const iconItems: Array<{ icon: React.ReactElement; text: string }> = []
+  if (address) {
+    iconItems.push({
+      icon: <FaAddressCard className="icon-list-icon" />,
+      text: address,
+    })
+  }
+
+  if (type && type.value) {
+    iconItems.push({
+      icon: <FaStreetView className="icon-list-icon" />,
+      text: type.value,
+    })
+  }
+
+  iconItems.push({
+    icon: <FaClock className="icon-list-icon" />,
+    text: appointment.cancelled ? 'Appointment cancelled' : `${start} - ${end}`,
+  })
+  return iconItems
+}
+
+export type AppointmentTileProps = {
+  appointment: ExtendedAppointmentModel
+  appointmentTypes: ListItemModel[]
+}
+
+export const AppointmentTile: React.FC<AppointmentTileProps> = ({ appointment, appointmentTypes }) => {
```

Comment:
If `AppointmentTile` and `AppointmentList` become bigger in the future, it can make this file huge. So keeping components in separate files can be preferred for the better organization.